### PR TITLE
kosu-genesis-cli: adding package

### PR DESCRIPTION
**Pull request by hrharder**
## Overview

Adding utility (`gen-kosu`, package [`@kosu/genesis-cli`](https://github.com/ParadigmFoundation/kosu-monorepo/tree/feature/kosu-genesis-cli/packages/kosu-genesis-cli)) to generate network genesis file from contract system state.

[_Click here to filter diff for source/test files only._](https://github.com/ParadigmFoundation/kosu-monorepo/pull/280/files?file-filters%5B%5D=.json&file-filters%5B%5D=.ts)

## Details
[See package docs for more info.](https://github.com/ParadigmFoundation/kosu-monorepo/tree/feature/kosu-genesis-cli/packages/kosu-genesis-cli/docs)

### CODE REVIEW

File: packages/kosu-genesis-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  -V, --version                          output the version number
+  -n, --chain-id <name>                  Specify the resulting Kosu chain ID
+  -p, --provider-url <url>               HTTP Ethereum JSONRPC provider
+  -b, --snapshot-block <number>          The block height at which to snapshot Kosu contract system state
+  -t, --start-time <number>              Unix timestamp (in seconds) of network start
+  -f, --finality-threshold <number>      Minimum age of Ethereum blocks before submitting attestations (default: "10")
+  -l, --period-limit <number>            Maximum number of order messages to accept per rebalance period (default: "100000")
+  -L, --period-length <number>           The length of each rebalance period (in Ethereum blocks) (default: "5")
+  -m, --max-order-bytes <number>         The maximum size of a single order transaction (default: "4096")
+  -c, --confirmation-threshold <number>  Amount of network vote power needed to accept witness events (updated each block) (default: "0")
```
Comment (gchaincl) [inline packages/kosu-genesis-cli/README.md]: now that confirmation threshold has been removed from the genesis block, do we still need this flag?

File: packages/kosu-genesis-cli/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    -n, --chain-id <name>
+    -p, --provider-url <url>
+    -b, --snapshot-block <number>
+    -t, --start-time <number>
+
+Options:
+  -V, --version                          output the version number
+  -n, --chain-id <name>                  Specify the resulting Kosu chain ID
+  -p, --provider-url <url>               HTTP Ethereum JSONRPC provider
+  -b, --snapshot-block <number>          The block height at which to snapshot Kosu contract system state
```
Comment (gchaincl) [inline packages/kosu-genesis-cli/README.md]: eventually, will we provide a default value for this?

File: packages/kosu-genesis-cli/src/cli/main.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        kosu = new Kosu({ provider });
+    } catch (error) {
+        throw new Error(`Failed to setup provider: ${error}`);
+    }
+
+    try {
+        const genesis = await generateGenesisFromBlock(
+            kosu,
+            chainId,
+            parseInt(snapshotBlock),
```
Comment (gchaincl) [inline packages/kosu-genesis-cli/src/cli/main.ts]: if we pass (ie string) to `parseInt` it returns `NaN`, as `generateGenesisFromBlock` receives a `number` would that be interpreted as a zero?

File: packages/kosu-genesis-cli/src/functions.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    for (const validator of validators) {
+        genesisValidators.push({
+            address: publicKeyToAddress(validator.publicKey),
+            pub_key: {
+                type: "tendermint/PubKeyEd25519",
+                value: validator.publicKey.toString("base64"),
+            },
+
+            // Note: power is calculated and set by the network
+            power: "0",
```
Comment (gchaincl) [inline packages/kosu-genesis-cli/src/functions.ts]: If we don't specify power, the client will assume it's zero, is there any reason to explicitly use `0`?

File: packages/kosu-genesis-cli/README.md (hunk 1 above)
Comment (hrharder) [reply packages/kosu-genesis-cli/README.md]: Thank you! have a commit i need to push with this

File: packages/kosu-genesis-cli/src/functions.ts (hunk 4 above)
Comment (hrharder) [reply packages/kosu-genesis-cli/src/functions.ts]: Nope -- I wasn't aware we could omit it. I will do that.

File: packages/kosu-genesis-cli/src/cli/main.ts (hunk 3 above)
Comment (hrharder) [reply packages/kosu-genesis-cli/src/cli/main.ts]: Not sure I fully understand the question, but what's happening is:
- `generateGenesisFromBlock` receives snapshot block as a `number`
- the snapshotBlock we get from the CLI is a string (but we do validation in cli/main.ts) so we `parseInt` it
- TypeScript also prevents us from passing a string (not if someone is using JS, of course) so there is that safety too

File: packages/kosu-genesis-cli/README.md (hunk 1 above)
Comment (hrharder) [reply packages/kosu-genesis-cli/README.md]: Looks like it was done, but the docs weren't updated

**Pull request merged**